### PR TITLE
[User History] Don't translate messages

### PR DESCRIPTION
**Pull request by mkangia**
## Summary

This PR skips translating messages before storing them for User History.
At some point we would need to parse back these messages, which would not be possible if they are translated.
If translation is needed, it should be handled at the UI level and not the message being stored.

[User History Feature](https://github.com/dimagi/commcare-hq/pull/29887) was merged on July 12th.

Translations updated last in master with [commit](https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b)

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

There are tests for importer changes.

### Safety story

Simple changes.
I just need to confirm that there has not been any translations yet. open to suggestions if any. Considering checking django translation files that if a translation for any of the string was added

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by snopoke — COMMENTED** [review]
Would you not still need to mark these for translation using `gettext_noop` otherwise there will be no translations for them.

File: corehq/apps/domain/models.py [hunk 1]
```diff
@@ -994,7 +994,7 @@ def transfer_domain(self, by_user, *args, transfer_via=None, **kwargs):
         if by_user:
             log_user_change(self.domain, couch_user=self.from_user,
                             changed_by_user=by_user, changed_via=transfer_via,
-                            message=_("Removed from domain '{domain_name}'").format(domain_name=self.domain))
+                            message=f"Removed from domain '{self.domain}'")
```
Comment (mkangia) [inline corehq/apps/domain/models.py:997]: newly added this week
https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R14386

File: corehq/apps/user_importer/helpers.py [hunk 2]
```diff
@@ -130,10 +129,7 @@ def save_log(self):
         if self.role_updated:
             new_role = self.user.get_role(domain=self.user_domain)
             if new_role:
-                self.logger.add_info(_("Role: {new_role_name}[{new_role_id}]").format(
-                    new_role_name=new_role.name,
-                    new_role_id=new_role.get_qualified_id()
-                ))
+                self.logger.add_info(f"Role: {new_role.name}[{new_role.get_qualified_id()}]")
```
Comment (mkangia) [inline corehq/apps/user_importer/helpers.py:132]: newly added
https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R32284-R32287

File: corehq/apps/user_importer/helpers.py [hunk 3]
```diff
@@ -149,14 +145,12 @@ def _include_user_data_changes(self):
 class CommCareUserImporter(BaseUserImporter):
     def update_password(self, password):
         self.user.set_password(password)
-        self.logger.add_change_message(_("Password Reset"))
+        self.logger.add_change_message("Password Reset")
```
Comment (mkangia) [inline corehq/apps/user_importer/helpers.py:148]: was already present
https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R16207

File: corehq/apps/user_importer/helpers.py [hunk 4]
```diff
@@ -149,14 +145,12 @@ def _include_user_data_changes(self):
...
 
     def update_phone_number(self, phone_number):
         fmt_phone_number = _fmt_phone(phone_number)
         if fmt_phone_number not in self.user.phone_numbers:
-            self.logger.add_change_message(_("Added phone number {new_phone_number}").format(
-                new_phone_number=fmt_phone_number)
-            )
+            self.logger.add_change_message(f"Added phone number {fmt_phone_number}")
```
Comment (mkangia) [inline corehq/apps/user_importer/helpers.py:153]: newly added
https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R32289

File: corehq/apps/user_importer/helpers.py [hunk 5]
```diff
@@ -190,7 +184,7 @@ def update_user_data(self, data, uncategorized_data, profile, domain_info):
                 self.user.pop_metadata(key)
 
         if self.user.user_data.get(PROFILE_SLUG) and self.user.user_data[PROFILE_SLUG] != current_profile_id:
-            self.logger.add_info(_("CommCare Profile: {profile_name}").format(
+            self.logger.add_info("CommCare Profile: {profile_name}".format(
```
Comment (mkangia) [inline corehq/apps/user_importer/helpers.py:187]: newly added
https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R32294

File: corehq/apps/user_importer/helpers.py [hunk 6]
```diff
@@ -226,19 +220,14 @@ def update_locations(self, location_codes, domain_info):
             if location_ids:
                 location_names = [get_location_from_site_code(code, domain_info.location_cache).name
                                   for code in location_codes]
-                self.logger.add_info(_("Assigned locations: {location_names}").format(
-                    location_names=location_names
-                ))
+                self.logger.add_info(f"Assigned locations: {location_names}")
```
Comment (mkangia) [inline corehq/apps/user_importer/helpers.py:223]: https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R32299

File: corehq/apps/user_importer/helpers.py [hunk 7]
```diff
@@ -226,19 +220,14 @@ def update_locations(self, location_codes, domain_info):
...
         # log this after assigned locations are updated, which can re-set primary location
         if self.user.location_id != user_current_primary_location_id:
             self.logger.add_changes({'location_id': self.user.location_id})
             if self.user.location_id:
                 user_updated_primary_location_name = get_user_primary_location_name(self.user, self.user_domain)
-                self.logger.add_info(
-                    _("Primary location: {primary_location_name}").format(
-                        primary_location_name=user_updated_primary_location_name
-                    ))
+                self.logger.add_info(f"Primary location: {user_updated_primary_location_name}")
```
Comment (mkangia) [inline corehq/apps/user_importer/helpers.py:230]: newly added
https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R32304

File: corehq/apps/user_importer/helpers.py [hunk 8]
```diff
@@ -252,19 +241,13 @@ def add_to_domain(self, role_qualified_id, location_id):
         self.user.add_as_web_user(self.user_domain, role=role_qualified_id, location_id=location_id)
         self.role_updated = bool(role_qualified_id)
 
-        self.logger.add_change_message(_("Added as web user to domain '{domain_name}'").format(
-            domain_name=self.user_domain)
-        )
+        self.logger.add_change_message(f"Added as web user to domain '{self.user_domain}'")
```
Comment (mkangia) [inline corehq/apps/user_importer/helpers.py:244]: newly added
https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R32309

File: corehq/apps/user_importer/helpers.py [hunk 9]
```diff
@@ -252,19 +241,13 @@ def add_to_domain(self, role_qualified_id, location_id):
...
             self._log_primary_location_info()
 
     def _log_primary_location_info(self):
         primary_location = self.user.get_sql_location(self.user_domain)
-        self.logger.add_info(
-            _("Primary location: {primary_location_name}[{primary_location_id}]").format(
-                primary_location_name=primary_location.name,
-                primary_location_id=primary_location.location_id
-            ))
+        self.logger.add_info(f"Primary location: {primary_location.name}[{primary_location.location_id}]")
```
Comment (mkangia) [inline corehq/apps/user_importer/helpers.py:250]: newly added

File: corehq/apps/user_importer/helpers.py [hunk 10]
```diff
@@ -300,17 +283,15 @@ def update_locations(self, location_codes, membership, domain_info):
                 locations_info = ", ".join([f"{location.name}[{location.location_id}]" for location in locations])
             else:
                 locations_info = []
-            self.logger.add_info(_("Assigned locations: {locations_info}").format(
-                locations_info=locations_info
-            ))
+            self.logger.add_info(f"Assigned locations: {locations_info}")
```
Comment (mkangia) [inline corehq/apps/user_importer/helpers.py:286]: newly added
https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R32319

File: corehq/apps/user_importer/helpers.py [hunk 11]
```diff
@@ -300,17 +283,15 @@ def update_locations(self, location_codes, membership, domain_info):
...
 
         # log this after assigned locations are updated, which can re-set primary location
         user_updated_primary_location_id = get_user_primary_location_id(self.user, self.user_domain)
         if user_updated_primary_location_id != user_current_primary_location_id:
             if user_updated_primary_location_id:
                 self._log_primary_location_info()
             else:
-                self.logger.add_info(_("Primary location: None"))
+                self.logger.add_info("Primary location: None")
```
Comment (mkangia) [inline corehq/apps/user_importer/helpers.py:294]: already present
https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R32324

File: corehq/apps/user_importer/importer.py [hunk 12]
```diff
@@ -306,7 +306,7 @@ def create_or_update_web_user_invite(email, domain, role_qualified_id, upload_us
     if invite_created and send_email:
         invite.send_activation_email()
     if invite_created and user_change_logger:
-        user_change_logger.add_change_message(_("Invited to domain '{domain_name}'").format(domain_name=domain))
+        user_change_logger.add_change_message(f"Invited to domain '{domain}'")
```
Comment (mkangia) [inline corehq/apps/user_importer/importer.py:309]: newly added

https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R32353

File: corehq/apps/user_importer/importer.py [hunk 13]
```diff
@@ -775,9 +775,7 @@ def remove_web_user_from_domain(domain, user, username, upload_user, user_change
         if is_web_upload:
             remove_invited_web_user(domain, username)
             if user_change_logger:
-                user_change_logger.add_change_message(_("Invitation revoked for domain '{domain_name}'").format(
-                    domain_name=domain
-                ))
+                user_change_logger.add_change_message(f"Invitation revoked for domain '{domain}'")
```
Comment (mkangia) [inline corehq/apps/user_importer/importer.py:778]: newly added 
https://github.com/dimagi/commcare-hq/commit/c7e498725df7a04c5b8133fbbaa750c6fbe3486b#diff-c9557c19413e04ae10a25832b30f73ee68d4e9bdae31b49f5404b9086b80f126R32410

File: corehq/apps/user_importer/helpers.py (hunk 3 above)
Comment (mkangia) [reply corehq/apps/user_importer/helpers.py:148]: Not sure why the en file has
```
msgid "Password Reset"
msgstr "Manejo de Casos"
```

File: corehq/apps/user_importer/helpers.py (hunk 11 above)
Comment (mkangia) [reply corehq/apps/user_importer/helpers.py:294]: actually this was newly added added but a fuzzy match populated
its surprising, like password reset, even the en file has
```
#: corehq/apps/user_importer/helpers.py:313
#, fuzzy
msgid "Primary location: None"
msgstr "Casos IRA Tratados"
```

**Comment by mkangia** [comment]
Hey @snopoke 

> Would you not still need to mark these for translation using gettext_noop otherwise there will be no translations for them.

I am thinking that UI would need to handle translation entirely and going to leave that for later. I want to just make sure to get this in asap so that the data in tables is in english.

The way I see that there would be a parser which would split and read messages and then identify them based on the structure of the message to translate it accordingly

```
def parse_message(change_message):
    messages = change_message.split(.)
    for message in messages:
        if is_role_change(message):
            role_name, role_id = get_role_name_and_id(message)
            _("Role changed to {role_name}" # this could be extended to have a link to the role page
        elif is_location_change(message):
            # ...
```

File: corehq/apps/user_importer/helpers.py (hunk 3 above)
Comment (mkangia) [reply corehq/apps/user_importer/helpers.py:148]: en
```
#: corehq/apps/domain/templates/login_and_password/password_reset_form.html:7
#: corehq/apps/domain/urls.py:87 corehq/apps/user_importer/helpers.py:152
#, fuzzy
msgid "Password Reset"
msgstr "Manejo de Casos"
```

es
```

#: corehq/apps/domain/templates/login_and_password/password_reset_form.html:7
#: corehq/apps/domain/urls.py:87 corehq/apps/user_importer/helpers.py:152
msgid "Password Reset"
msgstr "Contraseña Restablecida"
```

fra
```
#: corehq/apps/domain/templates/login_and_password/password_reset_form.html:7
#: corehq/apps/domain/urls.py:87 corehq/apps/user_importer/helpers.py:152
msgid "Password Reset"
msgstr "Réinitialisation du mot de passe"
```

no translations in other langs

File: corehq/apps/user_importer/helpers.py (hunk 11 above)
Comment (mkangia) [reply corehq/apps/user_importer/helpers.py:294]: en
```
#: corehq/apps/user_importer/helpers.py:313
#, fuzzy
msgid "Primary location: None"
msgstr "Casos IRA Tratados"
```

fra
```
#: corehq/apps/user_importer/helpers.py:313
#, fuzzy
#| msgid "Primary Location"
msgid "Primary location: None"
msgstr "Site principal"
```

por
```
#: corehq/apps/user_importer/helpers.py:313
#, fuzzy
#| msgid "Primary Location"
msgid "Primary location: None"
msgstr "Localização Primária"
```

es
```
#: corehq/apps/user_importer/helpers.py:313
#, fuzzy
#| msgid "Primary Location"
msgid "Primary location: None"
msgstr "Ubicación Principal"
```

no translation in other langs

**Comment by mkangia** [comment]
in an effort to find out if its safe to make this change now and that nothing was translated till now,
left a comment pointing to the change in HQ locale "en" file.

consolidating that info,
So there were two strings
"Password Reset" and "Primary Location: None"
both used by user importer, that had a translation.

Thinking about it now that if it would have even gotten translated at the first place because the user importer code is invoked through a task in celery? so we are okay here?

**Comment by snopoke** [comment]
> Hey @snopoke
> 
> > Would you not still need to mark these for translation using gettext_noop otherwise there will be no translations for them.
> 
> I am thinking that UI would need to handle translation entirely and going to leave that for later. I want to just make sure to get this in asap so that the data in tables is in english.
> 
> The way I see that there would be a parser which would split and read messages and then identify them based on the structure of the message to translate it accordingly
> 
> ```
> def parse_message(change_message):
>     messages = change_message.split(.)
>     for message in messages:
>         if is_role_change(message):
>             role_name, role_id = get_role_name_and_id(message)
>             _("Role changed to {role_name}" # this could be extended to have a link to the role page
>         elif is_location_change(message):
>             # ...
> ```

That does seem like a good way to do it but also means that we shouldn't be storing 'messages' and only store structured data.

**Comment by mkangia** [comment]
> That does seem like a good way to do it but also means that we shouldn't be storing 'messages' and only store structured data.

@snopoke , There are few things that would probably stay as messages like "Password Reset" or "Added to domain 'XYZ'", or possibly can be made into structured data as well if needed. But things like "Role: role-name[role-id]" could definitely be given more structure. 

It's possible this would go into some re-structure in future and then would need a migration. But before that it's important that the messages are predictable, which is what this PR is trying to achieve.

More context:
had a long call with Clayton yesterday, about this feature and also the future of this, and we were talking about how this can be extended to other models as well, hence should have a structure that is generic and not very user specific.
His vision is like the activity feed one sees when they visit https://github.com/
So, follow up is to think about design of this overall again and then based on that, migrate/re-start logs, hopefully just migrate.

Let me know if that helps, happy to offline with more specifics.

**Comment by mkangia** [comment]
> Thinking about it now that if it would have even gotten translated at the first place because the user importer code is invoked through a task in celery? so we are okay here?

https://docs.celeryproject.org/projects/django-celery/en/2.4/faq.html#generating-a-template-in-a-task-doesn-t-seem-to-respect-my-i18n-settings

This would mean it would not have translated anyway. So we should have the raw english message in all logs till now.

**Comment by snopoke** [comment]
> > That does seem like a good way to do it but also means that we shouldn't be storing 'messages' and only store structured data.
> 
> @snopoke , There are few things that would probably stay as messages like "Password Reset" or "Added to domain 'XYZ'", or possibly can be made into structured data as well if needed. But things like "Role: role-name[role-id]" could definitely be given more structure.

For these 2 example I would think something like `{"password": "reset"}` (where 'reset' is a slug and not what actually get's displayed, `{"domains": {"added": ["XYZ"]}`, {"role": {"id": role_id, "name": role_name}}`

> 
> It's possible this would go into some re-structure in future and then would need a migration. 

That seems like a lot of extra effort. Is it not worth pivoting the current design even if it means wiping any existing messages (which there can't be that many of since it has only just been released)

Overall I'm fine with this PR but I think it would be good to try and get this right before we release the feature. It doesn't seem like it would be that much more work if we just changed it to be more generic and to record structured data.

**Comment by mkangia** [comment]
Hey @snopoke 

> For these 2 example I would think something like {"password": "reset"} (where 'reset' is a slug and not what actually get's displayed, {"domains": {"added": ["XYZ"]}, {"role": {"id": role_id, "name": role_name}}`

That definitely sounds like a way this could be done.
I would like to think about this more before making such a major change to the current implementation.
Would like to be sure first that it fits in what we want logging models to look like eventually so that we don't have to change it again.

> That seems like a lot of extra effort. Is it not worth pivoting the current design even if it means wiping any existing messages (which there can't be that many of since it has only just been released)

Just checked current records on prod.
Prod got 65,221 records in like a day (Wow! I guess this time its growing faster than before because we are also tracking bulk imports). 
Agreed that losing/storing separately a day of records is better than doing a migration later, if that can be avoided. But i am not certain of that yet. So don't want to rush to changing implementation right away.

> Overall I'm fine with this PR but I think it would be good to try and get this right before we release the feature. It doesn't seem like it would be that much more work if we just changed it to be more generic and to record structured data.

I think this feature itself was in a good state and hence was released. Covering other models or making this generic was not part of the plan earlier and is still in very early stages.

**Review by orangejenny — APPROVED** [review]
Can you add comments explaining why these aren't translated? Maybe in `add_info`/`log_user_change`/`add_change_message`.

Can you think of any enforcement mechanism? I haven't been able to think of anything other than being diligent in code review. I think it's going to be easy to inadvertently add translations - my brain is very much trained to flag any user-friendly string as needing translation.

> If translation is needed, it should be handled at the UI level and not the message being stored.

Are you imagining something in particular here? I would think we'd need to store both a translated and an untranslated version at the time the message is created.

> I just need to confirm that there has not been any translations yet.

This has only been live for a couple of days and it's still under development. If confirming this is going to be annoying, I think you'd be fine to just delete any existing `UserHistory` objects.

**Comment by orangejenny** [comment]
Sorry for repeating points, I just looked at the code and missed the conversation on the PR.

> Overall I'm fine with this PR but I think it would be good to try and get this right before we release the feature. It doesn't seem like it would be that much more work if we just changed it to be more generic and to record structured data.

I agree with this. This PR is good so that we stop storing translations for now, but I think it's worth the time to consider storing messages in a more structured format. Trying to parse info back out of unstructured messages gets messy - I thought about doing it for the property filter in https://github.com/dimagi/commcare-hq/pull/30058, for the user profiles, but it's very fragile.

> I think this feature itself was in a good state and hence was released.

I agree it's in a good state, but I'm not certain it's fully finished. There's still a lot of freedom to make changes now - and to do things like wipe the UserHistory table - and it'll get much harder to so once we give access to external users.

**Comment by mkangia** [comment]
> Would you not still need to mark these for translation using gettext_noop otherwise there will be no translations for them.

@snopoke I believe these would get added back for translations once they are marked for translations for UI?
We don't have translations for them yet so we should not be losing any translations.

**Comment by mkangia** [comment]
@orangejenny 

> Can you add comments explaining why these aren't translated? Maybe in add_info/log_user_change/add_change_message.

Sure

> > If translation is needed, it should be handled at the UI level and not the message being stored.

> Are you imagining something in particular here? I would think we'd need to store both a translated and an untranslated version at the time the message is created.

I imagine something like https://github.com/dimagi/commcare-hq/pull/30064#issuecomment-881326255. Why do you say we should store both translated and untranslated version?

> This has only been live for a couple of days and it's still under development. If confirming this is going to be annoying, I think you'd be fine to just delete any existing UserHistory objects.

I could check all po files, and could confirm that there was no translation already or if there were, they were not being used. So that is all cool.


>  Trying to parse info back out of unstructured messages gets messy - I thought about doing it for the property filter in #30058, for the user profiles, but it's very fragile.

I meant for parsing text messages only for translations. but still in favor of a structured message like Simon suggested in his comment.
The filter part you mentioned in the linked PR, that was related to filtering only right? It would still be true if in a structured message that says `'profile': 'abc'` or `'user_data': {'profile': 'abc'}`, where the latter is what the current implementation should have?

> There's still a lot of freedom to make changes now - and to do things like wipe the UserHistory table - and it'll get much harder to change once we give access to external users.

Ya. I believe there is still a lot of time before this gets handed out to external users so it won't come to that point.
Though I am not in favor of wiping out the table at any point unless there is no way to migrate data or if the migration gets too complex.

**Comment by mkangia** [comment]
@orangejenny 
added https://github.com/dimagi/commcare-hq/pull/30064/commits/e790478f566d091512ff09e5fa134001383512cf

and not saying that this is full proof but added it for now.

**Comment by snopoke** [comment]
> > There's still a lot of freedom to make changes now - and to do things like wipe the UserHistory table - and it'll get much harder to change once we give access to external users.
> 
> Ya. I believe there is still a lot of time before this gets handed out to external users so it won't come to that point.
> Though I am not in favor of wiping out the table at any point unless there is no way to migrate data or if the migration gets too complex.

+1 to getting this right before releasing it. If a migration is easy that's fine but if it's going to be a lot more work I'd also be in favour of dropping existing data.

**Comment by mkangia** [comment]
thanks @simon @orangejenny 

Offlined with Clayton and we were aligned on changing to use structured messages like Simon suggested. Will add a ticket and start working on it.

> If a migration is easy that's fine but if it's going to be a lot more work I'd also be in favour of dropping existing data.

Got it. Will start on writing one with the changes. If I get a sense it's not worth the effort, will drop it.

Going to merge this PR though, just waiting on build

**Comment by mkangia** [comment]
Hey @snopoke @orangejenny 

Added [this doc](https://docs.google.com/document/d/15WS2OuB8l8UJePuwnZASIerxIrB2Gfnf1RRt24O3n8M/edit#) to spec the changes out for structured messages

Please share thoughts/suggestions.

**Comment by mkangia** [comment]
Strange finding,

> Prod got 65,221 records in like a day (Wow! I guess this time its growing faster than before because we are also tracking bulk imports).

which was 6 days ago, and the count of records as of now on prod is  72,725.
I could not pin down a domain or bulk upload that had a lot of updates. Guess the user activity on first day of release was really high.

**Pull request merged**